esdm/esdm_drng_mgr.c

```diff
@@ -664,10 +668,10 @@ void esdm_drng_force_reseed(void)
 static bool esdm_drng_must_reseed(struct esdm_drng *drng)
 {
 	struct timespec check_time = drng->last_seeded;
-	bool internal_bits_reached = (ESDM_DRNG_RESEED_THRESH_BITS > 0) && (atomic_read(&drng->internal_bits) >= ESDM_DRNG_RESEED_THRESH_BITS);
+	bool request_bits_since_fully_seeded_reached = (ESDM_DRNG_RESEED_THRESH_BITS != UINT32_MAX) && (atomic_read_u32(&drng->request_bits_since_fully_seeded) >= ESDM_DRNG_RESEED_THRESH_BITS);
```

Comment:
What about simply removing the check for UINT32_MAX?

Reply:
I will implement this in the next round. See comment above.

Reply:
The current code removed the UINT32_MAX check, but as I also noticed, that the condition needs to be >= instead of >, without the check for UINT32_MAX this cannot be fully disabled anymore. If you can live with reseeding after at most 512 MB per DRNG we can leave it as is, otherwise the UINT32_MAX has to be reintroduced.